connect.go

```diff
@@ -31,18 +31,19 @@ const (
 )
 
 func (c *baseConn) connect(r *buff.Reader, cfg *connConfig) error {
-	c.writer.BeginMessage(message.ClientHandshake)
-	c.writer.PushUint16(0) // major version
-	c.writer.PushUint16(8) // minor version
-	c.writer.PushUint16(2) // number of parameters
-	c.writer.PushString("database")
-	c.writer.PushString(cfg.database)
-	c.writer.PushString("user")
-	c.writer.PushString(cfg.user)
-	c.writer.PushUint16(0) // no extensions
-	c.writer.EndMessage()
-
-	if err := c.writer.Send(c.conn); err != nil {
+	w := buff.NewWriter(c.writeMemory[:0])
```

Comment:
What does `c.writeMemory[:0]` do? Returns a pointer to the first byte of the buffer?

Reply:
Basically, yes. `NewWriter()` takes a `[]byte` with `len(bytes) == 0` but expects it to have allocated capacity. `c.writeMemory[:0]` creates a zero length slice with allocated capacity equal to the length of the `writeMemory` array.